pkg/packager/ocilayout.go

```diff
@@ -0,0 +1,49 @@
+package packager
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pivotal/image-relocation/pkg/image"
+	"github.com/pivotal/image-relocation/pkg/registry"
+)
+
+// ociLayout is an ImageStore which stores images as an OCI image layout.
+type ociLayout struct {
+	registryClient registry.Client
+
+	layout registry.Layout
+	logs   io.Writer
+}
+
+func newOciLayout() *ociLayout {
+	return &ociLayout{
+		registryClient: registry.NewRegistryClient(),
+	}
+}
+
+func (t *ociLayout) configure(archiveDir string, logs io.Writer) error {
+	layoutDir := filepath.Join(archiveDir, "artifacts", "layout")
+	if err := os.MkdirAll(layoutDir, 0755); err != nil {
+		return err
+	}
+
+	layout, err := t.registryClient.NewLayout(layoutDir)
+	if err != nil {
+		return err
+	}
+
+	t.layout = layout
+	t.logs = logs
+	return nil
+}
+
+func (t *ociLayout) add(im string) error {
+	n, err := image.NewName(im)
+	if err != nil {
+		return err
+	}
+	_, err = t.layout.Add(n)
```

Comment:
We should return the digest and ensure that the image has a matching digest in the exported bundle. This will enable the image to be found in the OCI image layout, e.g. during thick bundle relocation.

Reply:
Decided that was too much for this PR. For now, we simply fail if non-empty digests do not match.

Reply:
LGTM to defer this work.
Is #408 supposed to track this?

Reply:
Yes indeed. See the questions I raised: https://github.com/deislabs/duffle/issues/408#issuecomment-486612458